NorthstarDLL/plugins/plugins.cpp

```diff
@@ -54,116 +55,69 @@ std::optional<Plugin> PluginManager::LoadPlugin(fs::path path, PluginInitFuncs*
 	Plugin plugin {};
 
 	std::string pathstring = path.string();
-	std::wstring wpath = path.wstring();
+	std::wstring wpath = path.wstring(); // is there any reason to load the libs with wide strings?
 
 	LPCWSTR wpptr = wpath.c_str();
-	HMODULE datafile = LoadLibraryExW(wpptr, 0, LOAD_LIBRARY_AS_DATAFILE | LOAD_LIBRARY_AS_IMAGE_RESOURCE); // Load the DLL as a data file
-	if (datafile == NULL)
-	{
-		NS::log::PLUGINSYS->info("Failed to load library '{}': ", std::system_category().message(GetLastError()));
-		return std::nullopt;
-	}
-	HRSRC manifestResource = FindResourceW(datafile, MAKEINTRESOURCEW(IDR_RCDATA1), RT_RCDATA);
 
-	if (manifestResource == NULL)
-	{
-		NS::log::PLUGINSYS->info("Could not find manifest for library '{}'", pathstring);
-		freeLibrary(datafile);
-		return std::nullopt;
-	}
-	HGLOBAL myResourceData = LoadResource(datafile, manifestResource);
-	if (myResourceData == NULL)
+	HMODULE pluginLib =
+		LoadLibraryExW(wpptr, 0, LOAD_LIBRARY_SEARCH_USER_DIRS | LOAD_LIBRARY_SEARCH_DEFAULT_DIRS); // Load the DLL with lib folders
+
+	if (!pluginLib)
 	{
-		NS::log::PLUGINSYS->error("Failed to load manifest from library '{}'", pathstring);
-		freeLibrary(datafile);
+		NS::log::PLUGINSYS->error("'Failed to load library '{}'", std::system_category().message(GetLastError())); // GetLastError??? We know the path that failed to load
 		return std::nullopt;
 	}
-	int manifestSize = SizeofResource(datafile, manifestResource);
-	std::string manifest = std::string((const char*)LockResource(myResourceData), 0, manifestSize);
-	freeLibrary(datafile);
 
-	rapidjson_document manifestJSON;
-	manifestJSON.Parse(manifest.c_str());
+	uint32_t* abiVersion = (uint32_t*)GetProcAddress(pluginLib, "PLUGIN_ABI_VERSION");
+	char** name = (char**)GetProcAddress(pluginLib, "PLUGIN_NAME");
+	char** description = (char**)GetProcAddress(pluginLib, "PLUGIN_DESCRIPTION");
+	char** abbreviation = (char**)GetProcAddress(pluginLib, "PLUGIN_ABBREVIATION");
+	char** version = (char**)GetProcAddress(pluginLib, "PLUGIN_VERSION");
+	char** dependencyName = (char**)GetProcAddress(pluginLib, "PLUGIN_DEPENDENCY_NAME");
+	uint8_t* context = (uint8_t*)GetProcAddress(pluginLib, "PLUGIN_CONTEXT");
+	PLUGIN_INIT_TYPE plugin_init = (PLUGIN_INIT_TYPE)GetProcAddress(pluginLib, "PLUGIN_INIT");
```

Comment:
since these are exported symbols, they are likely read only and should be marked as const so the compiler knows we shouldn't be doing anything funny.